lightning/src/ln/channel.rs

```diff
@@ -736,6 +736,9 @@ pub(super) struct Channel<Signer: Sign> {
 	// don't currently support node id aliases and eventually privacy should be provided with
 	// blinded paths instead of simple scid+node_id aliases.
 	outbound_scid_alias: u64,
+
+	// We track whether we already emitted a `ChannelReady` event.
+	channel_ready_event_emitted: bool,
```

Comment:
Is it a big deal if this is race-y? Would that ever happen?

Reply:
I think it would matter, as we may emit the event multiple times, which in turn could trigger some logic of the user depending on it. It shouldn't happen however, as we hold the lock on the channel for the duration of `emit_channel_ready_event!`. But, I now also moved taking and dropping the lock into the macro to ensure we really only setting it emitted after we pushed to the event queue and dropped its lock.